ccbuilder/builder/builder.py

```diff
@@ -41,6 +41,13 @@ class BuildContext:
     def __enter__(self) -> tuple[Path, TextIO]:
         self.build_dir = tempfile.mkdtemp()
         os.makedirs(self.install_prefix, exist_ok=True)
+        worker_pid_file = self.install_prefix / "WORKER_PID"
+
+        if worker_pid_file.exists():
+            raise BuildException(
+                "Compiler is either being built by another process "
+                "or the cache in a bad state (run ccbuilder cache clean to fix)."
+            )
```

Comment:
I think there should be a special exception for this. 
For example, currently, if I want to re-implement the waiting routine in a program using ccbuilder, how would I differentiate if there's just another worker or the build actually failed?

tbh I think having to wait for another worker is not exceptional behavior.